extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/security/HttpAuthenticator.java

```diff
@@ -293,13 +308,31 @@ private Uni<HttpAuthenticationMechanism> getPathSpecificMechanism(int index, Rou
                     public HttpAuthenticationMechanism apply(HttpCredentialTransport t) {
                         if (t != null && t.getAuthenticationScheme().equalsIgnoreCase(pathSpecificMechanism)) {
                             routingContext.put(HttpAuthenticationMechanism.class.getName(), mechanisms[index]);
+                            routingContext.put(AUTH_MECHANISM, t.getAuthenticationScheme());
                             return mechanisms[index];
                         }
                         return null;
                     }
                 });
     }
 
+    static void selectAuthMechanismWithAnnotation() {
+        selectAuthMechanismWithAnnotation = true;
+    }
+
+    static void selectAuthMechanism(RoutingContext routingContext, String authMechanism) {
+        if (requestAlreadyAuthenticated(routingContext, authMechanism)) {
+            throw new AuthenticationFailedException("""
+                    Request has been authenticated with different authentication mechanism before the '%s' authentication
```

Comment:
Propose to rephrase it slightly:

```
The '%s' authentication mechanism is required to authenticate the request but it was already authenticated with the '%s' authentication mechanism. It can happen if the '%s' is selected with an annotation but '%s' is activated by the HTTP security policy which is enforced before the JAX-RS chain is run. In such cases, please set the 'quarkus.http.auth.permission."permissions".applies-to=jaxrs' to all HTTP security policies which secure the same REST endpoints as the ones secured by the  '%s' authentication mechanism selected with the annotation.
```
which will have some `%s` duplications

Reply:
yes, that's much better, thanks